vault/resource_secrets_sync_association.go

```diff
@@ -89,48 +106,99 @@ func secretsSyncAssociationWrite(ctx context.Context, d *schema.ResourceData, me
 	}
 
 	log.Printf("[DEBUG] Writing association to %q", path)
-	resp, err := client.Logical().WriteWithContext(ctx, path, data)
+	_, err := client.Logical().WriteWithContext(ctx, path, data)
 	if err != nil {
 		return diag.Errorf("error setting secrets sync association %q: %s", path, err)
 	}
 	log.Printf("[DEBUG] Wrote association to %q", path)
 
-	// expect accessor to be provided from mount
-	accessor, err := getMountAccessor(ctx, d, meta)
-	if err != nil {
-		return diag.Errorf("could not obtain accessor from given mount; err=%s", err)
+	// ex: gh/dest/gh-dest-1/mount/kv/secret/token
+	// unique. each destination should only have one association for a particular accessor and secret
+	id := fmt.Sprintf("%s/dest/%s/mount/%s/secret/%s", destType, name, mount, secretName)
```

Comment:
This can be changed. Mainly need to format the information so that it can be extracted using a regex when needed. There are different precedents for doing this in the codebase, so happy to consider other alternatives

Reply:
Since it needs to be unique this sounds reasonable to me